**Vectores y valores propios de una matriz**
-
---

Cuando dos variables están altamente correlacionadas, quiere decir que aportan básicamente la misma información del problema y eso podría indicar que solo necesitamos una de ellas.

#### **Repaso de matrices**

<p align='justify'>
Las matrices son objetos matemáticos que tienen cierto número de filas y columnas (estos números son denominados dimensiones de la matriz). 

La transpuesta de una matriz es la operación en el que las filas cambian de lugar con las columnas y viceversa
</p>



<font size=3> $$ \left[\begin{matrix} 
4 & -3 \\
1 & 2
\end{matrix}\right]^T =
\left[\begin{matrix}
4 & 1 \\
-3 & 2
\end{matrix}\right] $$ </font>

#### **Indexación en una matriz**

<font size=3>$ A_{xy} $</font>

En la matriz $A$ encontraremos dos elemntos de indexación $x$ y $y$. Donde $x$ representa las columnas y $y$ las filas.

#### **Multiplicación de matrices**

In [51]:
import numpy as np
from fractions import Fraction

In [4]:
A = np.matrix([[2, 4], [-1,2]])
B = np.matrix([[3, 2], [6,-8]])
A, B

(matrix([[ 2,  4],
         [-1,  2]]),
 matrix([[ 3,  2],
         [ 6, -8]]))

In [5]:
## Multiplicación
np.matmul(A, B)

matrix([[ 30, -28],
        [  9, -18]])

#### **Inversa de una raíz**

In [8]:
## Usando A
A_inv = np.linalg.inv(A)
A_inv

matrix([[ 0.25 , -0.5  ],
        [ 0.125,  0.25 ]])

In [9]:
np.matmul(A, A_inv)

matrix([[1., 0.],
        [0., 1.]])

#### **Vectores**

Un vector es una matriz conformada por una sola columna, en la que cada fila representa la cantidad de dimensiones del vector.

Multiplicación de vectores:
$$
\vec{a}·\vec{v} = \begin{matrix}[a_{11} & a_{12}]\end{matrix}·\left[\begin{matrix}v_{11} \\ v_{21}\end{matrix}\right] = [c_{11}] = c_{11} = a_{11}v_{11} + a_{12}v_{21}
$$

#### **Vectores y valores propios de una matriz**

En álgebra lineal existen ecuaciones en las que la incógnita es un vector, como en la siguiente ecuación.

$$
A · \vec{x} = \lambda\vec{x}
$$

Aquí $A$ es una matriz cuadrada y $\vec{x}$ es un vector que desconocemos.

$$
A=\left[\begin{matrix}1&2 \\ 1&0\end{matrix}\right]
$$

$$
\left[\begin{matrix}1&2 \\ 1&0\end{matrix}\right]·\left[\begin{matrix}x\\y\end{matrix}\right] =  \lambda\left[\begin{matrix}x\\y\end{matrix}\right]
$$
$$
\left[\begin{matrix}x + 2y\\x\end{matrix}\right] = \left[\begin{matrix}\lambda{x}\\\lambda{y}\end{matrix}\right]
$$

Para satisfacer este sistema de ecuaciones se requiere de conceptos más detallados de algebra lineal.

En este caso se utilizará methodos de _numpy_ en _python_ para resolver este problema.

In [14]:
A = np.array([[1,2],[1,0]])
values, vectors = np.linalg.eig(A)
values, vectors

(array([ 2., -1.]),
 array([[ 0.89442719, -0.70710678],
        [ 0.4472136 ,  0.70710678]]))

$$
\left[\begin{matrix}
2 \\ -1
\end{matrix}\right]
\begin{cases}
\left[\begin{matrix}x\\y\end{matrix}\right] =
\left[\begin{matrix}0.89442719\\0.4472136\end{matrix}\right], \lambda=2 \\
\left[\begin{matrix}x\\y\end{matrix}\right] =
\left[\begin{matrix}-0.70710678\\0.70710678\end{matrix}\right], \lambda=-1
\end{cases}
$$

Para verificar que cada vector y su respectivo valor propio cumplen con la ecuación original se puede realizar una multiplicación de la matriz original y la primera fila de la transpuesta de la de su vectores.

$$ \left[\begin{matrix}1&2 \\ 1&0\end{matrix}\right]  · \left[\begin{matrix}0.89442719\\0.4472136\end{matrix}\right]^T$$

In [42]:
print(np.matmul(A, vectors.T[1]))
print((-1)*vectors.T[1])


[ 0.70710678 -0.70710678]
[ 0.70710678 -0.70710678]


In [40]:
print(values[1]*vectors.T[1])

[ 0.70710678 -0.70710678]


In [53]:
np.matmul(A, vectors)

array([[ 1.78885438,  0.70710678],
       [ 0.89442719, -0.70710678]])

El objetivo de esto es diagonalizar una matriz.

$$A=P·D·P^{-1}$$
<p align='center'>
Donde D es una matriz diagonal.
</p>

<p align='justify'>
Lo importante de estudiar este procedimiento en nuestro curso, es que cuando aplicamos este cálculo de vectores y valores propios a una matriz de covarianza, los vectores representan las direcciones a lo largo de las cuales percibimos la mayor cantidad de varianza de ese conjunto de datos, donde la cantidad de varianza es proporcional al valor propio de cada vector propio.

Y es importante tener en cuenta que este procedimiento aplica para un conjunto de datos con N variables al que le corresponde una matriz de covarianza de tamaño NxN.

Ahora, el último factor importante de esta técnica es que para matrices de covarianza, sus vectores propios siempre son independientes unos de otros y esto es justamente lo que queremos en un proceso de reducción de variables, porque direcciones independientes implica que estos vectores representan nuevas variables cuya correlación es la más baja posible y así cada nueva variable es lo más representativa posible.

En álgebra lineal se dice más precisamente que los vectores propios de una matriz de covarianza son ortogonales y esto quiere decir que el producto interno de cualquier par de estos vectores siempre da como resultado cero:
</p>

$$ \vec{x}_i·\vec{x}_j = 0 $$